### DISCUSSION TITLE
Solis Inverter. Query Inverter time

### DISCUSSION THREAD

**Discussion by spetrie13** [wills106/homeassistant-solax-modbus#725] [Ideas]
Would it be possible to configure the system to obtain the inverter time for Solis inverters?

**Comment by 0rangutan** [comment]
Can you expand on why you would like this?
I'm interested in a related capability which would be to have the integration sync the inverter's time on a regular (weekly?) basis to stop it drifting too far out.

**Reply by spetrie13** [reply]
Basically so that I can monitor the Inverter time and check it does not drift. No connection to cloud, so Inverter relies on internal clock only. I set times for charging / discharging for time of use tarrifs so need clock to be accurate to less than 1 difference. Use home assistant to view the time so I don't have to go to Inverter screen. I can sync from home assistant, but validate the Inverter time.

**Reply by wills106** [reply]
The Solis Inverter returns the RTC but it's [disabled by default](https://homeassistant-solax-modbus.readthedocs.io/en/latest/faq/#entity-i-need-is-not-present) as it will flood your log with useless data.

https://github.com/wills106/homeassistant-solax-modbus/blob/f76672cedc95cff2e675e50a1dfd5f4ca294b45c/custom_components/solax_modbus/plugin_solis.py#L788-L801

You are best use using an automation to trigger the button "Sync RTC" once a month of whatever frequency is required.
